### PR TITLE
feat: add ts-standard

### PR DESCRIPTION
**Pull request by zekth**
Following: https://github.com/fastify/fastify/pull/2699

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
There are still linting changes

**Comment by zekth** [comment]
@mcollina yes, i'm wondering about overrides. If we can't add overrides, we can fix types and create another issue to fix test linting errors after if we want to land it quicker.

File: test/types/plugin.test-d.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ expectAssignable<FastifyInstance>(fastify().register(testPluginCallback, {}))
 const testPluginAsync: FastifyPluginAsync = async function (instance, opts) { }
 expectAssignable<FastifyInstance>(fastify().register(testPluginAsync, {}))
 
-expectAssignable<FastifyInstance>(fastify().register(function (instance, opts): Promise<void> { return Promise.resolve() }))
+expectAssignable<FastifyInstance>(fastify().register(async function (instance, opts): Promise<void> { return await Promise.resolve() }))
```
Comment (fox1t) [inline test/types/plugin.test-d.ts:38]:
```suggestion
expectAssignable<FastifyInstance>(fastify().register(async function (instance, opts): Promise<void> { return Promise.resolve() }))
```

**Review by fox1t — CHANGES_REQUESTED** [review]

**Comment by mcollina** [comment]
there is no need for quick :)

**Comment by zekth** [comment]
I made some disables in files because `ts-standard` does not support the overrides, we use those for the tests.
Also i need help for the errors `void is only valid as a return type or generic type variable  @typescript-eslint/no-invalid-void-type` on the types. If someone got a lead on this.

**Comment by zekth** [comment]
@Ethan-Arrowood we are stuck on the last test with @fox1t , if you have a clue feel free to help

File: test/types/plugin.test-d.ts [hunk 2]
```diff
@@ -61,8 +62,8 @@ expectAssignable<PromiseLike<undefined>>(httpsServer.register(testPluginOptsWith
 expectAssignable<PromiseLike<undefined>>(httpsServer.register(testPluginOptsWithType, { prefix: '/test' }))
 expectAssignable<PromiseLike<undefined>>(httpsServer.register(testPluginOptsWithTypeAsync, { prefix: '/test' }))
 
-async function testAsync (): Promise<void> {
-  await httpsServer
-    .register(testPluginOpts)
-    .register(testPluginOpts)
-}
+// async function testAsync (): Promise<void> {
```
Comment (Ethan-Arrowood) [inline test/types/plugin.test-d.ts:65]: can this be removed if unused?

**Comment by Ethan-Arrowood** [comment]
huh im stumped. does it run locally by any chance? its strange that its the `put` one thats failing and not the others

**Comment by zekth** [comment]
> huh im stumped. does it run locally by any chance? its strange that its the `put` one thats failing and not the others

unfortunately no

**Comment by zekth** [comment]
@fastify/typescript any idea how we can fix and land this?

**Comment by Ethan-Arrowood** [comment]
> @fastify/typescript any idea how we can fix and land this?

I think if someone has the capacity, simplify this issue to a reproducible typescript playground link and we can go from there. I can forward it along to the TS folks if we think something is actually wrong

**Comment by zekth** [comment]
@fox1t i did a rebase and one more issue showed up :(

**Pull request closed without merging**